### PR TITLE
Ag 8617/waterfall total subtotal

### PR DESCRIPTION
**Pull request by manapeirov**
PR for https://ag-grid.atlassian.net/browse/AG-8617

### CODE REVIEW

File: packages/ag-charts-community/src/chart/axis/categoryAxis.ts [hunk 1]
```diff
@@ -35,12 +35,17 @@ export class CategoryAxis extends CartesianAxis<BandScale<string | object>> {
...
 
     normaliseDataDomain(d: (string | object)[]): (string | object)[] {
-        // Prevent duplicate categories.
-        const valuesSet: Record<string, (typeof d)[number]> = {};
-        for (const next of d) {
-            valuesSet[String(next)] ??= next;
+        const domain = [];
+        const uniqueValues = new Set();
+        for (const v of d) {
+            const key = v instanceof Date ? v.getTime() : v;
```
Comment (alexanderby) [inline packages/ag-charts-community/src/chart/axis/categoryAxis.ts:41]: Will it fail for `{ toString: () => 'abc' }`? Maybe it should be like `v instanceof Date ? v.getTime() : String(v)` or `v instanceof Date ? v.getTime() : typeof v === 'object' ? String(v) : v`.

Comment (alexanderby) [reply packages/ag-charts-community/src/chart/axis/categoryAxis.ts:41]: Nevermind 🤦

Comment (manapeirov) [reply packages/ag-charts-community/src/chart/axis/categoryAxis.ts:41]: no

Comment (alexanderby) [reply packages/ag-charts-community/src/chart/axis/categoryAxis.ts:41]: https://github.com/ag-grid/ag-charts/pull/67#discussion_r1280671452

**Review by alexanderby — APPROVED** [review]
🥇

**Pull request merged**